h2o-core/src/main/java/hex/grid/Grid.java

```diff
@@ -540,6 +535,68 @@ public void exportModelsBinary(final String exportDir) throws IOException {
     }
   }
 
+  @SuppressWarnings("rawtypes")
+  private Set<Key> getDependentKeysFromParams() {
```

Comment:
IMHO this code doesn't belong there, grid search shouldn't be itself responsible for its own recovery

instead, it should just say what objects or Keys are needed for resuming and propagate it to the layer above

we eventually want any process to be recoverable, I would expect GS to have a Recoverable interface that would include a method `Set<Key> getDependentObjects` - the layer avobe should be responsible for saving and re-loading those object upon restart

Reply:
refactored, please look it over